### DISCUSSION TITLE
Stateful icons for more entities.

### DISCUSSION THREAD

**Discussion by sebastianrasor** [home-assistant/frontend#17590] [Other feature requests]
I noticed that my fan entities have stateful icons, that is the icon is just a fan when the entity state is on, but the fan icon has a slash through it to indicate that the fan is off when the entity state is off.
Here are examples:
![image](https://github.com/home-assistant/frontend/assets/92653912/7e7dffb6-0ef2-490f-9774-a93eef358ed0)
![image](https://github.com/home-assistant/frontend/assets/92653912/e0ab4a80-dc5e-499f-b908-00a28d3f2e0f)
Note that I am NOT using any custom configuration for this. This is the default behavior of the fan entity (as well as some others as I've noticed, such as a binary_sensor with the type of garage door).

**Comment by sebastianrasor** [comment]
In addition, it'd be nice to be able to use templating anywhere you can define an icon so you can set an icon to be dependent on the state of another entity or use different stateful icons.

**Comment by Quentame** [comment]
Similar/duplicate from https://github.com/home-assistant/frontend/discussions/13700
